cocos/2d/assets/sprite-frame.ts

```diff
@@ -1438,9 +1439,28 @@ export class SpriteFrame extends Asset {
         sp._texture = this._texture;
         sp._isFlipUVX = this._isFlipUVX;
         sp._isFlipUVY = this._isFlipUVY;
+        if (this._original) {
+            sp._original = {
+                _texture: this._original._texture,
+                _x: this._original._x,
+                _y: this._original._y,
+            };
+        } else {
+            sp._original = null;
+        }
+        sp._packable = this._packable;
         sp._pixelsToUnit = this._pixelsToUnit;
         sp._pivot.set(this._pivot);
         sp._meshType = this._meshType;
+        sp._extrude = this._extrude;
+        sp._customOutLine.splice(0, sp._customOutLine.length, ...this._customOutLine);
+        sp._minPos = this._minPos;
+        sp._maxPos = this._maxPos;
+        if (this._mesh) {
+            // Creates a new mesh, and 'this' creates the mesh in the same way. So we can make a copy like this.
+            // It must be placed last because the mesh will depend on some of its members when it is created.
+            sp.ensureMeshData();
```

Comment:
The logic is not correct, `ensureMeshData ` will invoke `_initVertices` which will reset vertices to default values.

Reply:
It doesn't matter, the mesh and vertices are a combination. If this's mesh was generated by calling ensureMeshData, then sp should be the same in this way. The vertices and mesh will end up being the same, and it won't happen that the mesh is generated and then the vertices are modified.

Reply:
However, createmesh can be called to speed up the creation process